employee-rostering-frontend/src/store/availability/operations.ts

```diff
@@ -0,0 +1,96 @@
+ */
+
+import { ThunkCommandFactory } from '../types';
+import EmployeeAvailability from 'domain/EmployeeAvailability';
+import EmployeeAvailabilityView, { availabilityToAvailabilityView } from 'domain/EmployeeAvailabilityView';
+import moment from 'moment';
+import { alert } from 'store/alert';
+import { refreshShiftRoster, refreshAvailabilityRoster } from 'store/roster/operations';
+import { objectWithout } from 'util/ImmutableCollectionOperations';
+
+export interface KindaEmployeeAvailabilityView extends Omit<EmployeeAvailabilityView, "startDateTime" | "endDateTime"> {
+  startDateTime: string;
+  endDateTime: string;
+}
+
+export function availabilityAdapter(employeeAvailability: EmployeeAvailability): KindaEmployeeAvailabilityView {
+  return {
+    ...objectWithout(availabilityToAvailabilityView(employeeAvailability), "startDateTime", "endDateTime"),
+    startDateTime: moment(employeeAvailability.startDateTime).local().format("YYYY-MM-DDTHH:mm:ss"),
+    endDateTime: moment(employeeAvailability.endDateTime).local().format("YYYY-MM-DDTHH:mm:ss")
+  };
+}
+
+export function kindaAvailabilityViewAdapter(kindaAvailabilityView: KindaEmployeeAvailabilityView):
+EmployeeAvailabilityView {
+  return {
+    ...kindaAvailabilityView,
+    startDateTime: moment(kindaAvailabilityView.startDateTime).toDate(),
+    endDateTime: moment(kindaAvailabilityView.endDateTime).toDate(),
+  };
+}
+
+export const addEmployeeAvailability: ThunkCommandFactory<EmployeeAvailability, any> = employeeAvailability =>
+  (dispatch, state, client) => {
+    const tenantId = employeeAvailability.tenantId;
+    return client.post<KindaEmployeeAvailabilityView>(`/tenant/${tenantId}/employee/availability/add`,
+      availabilityAdapter(employeeAvailability)).then(newEmployeeAvailability => {
+      dispatch(alert.showSuccessMessage("addAvailability", { 
+        employeeName: employeeAvailability.employee.name, 
+        startDateTime: moment(employeeAvailability.startDateTime).format("LLL"),
+        endDateTime: moment(employeeAvailability.endDateTime).format("LLL") 
+      }));
+      dispatch(refreshShiftRoster());
+      dispatch(refreshAvailabilityRoster());
+    });
+  };
+
+export const removeEmployeeAvailability: ThunkCommandFactory<EmployeeAvailability, any> = 
+employeeAvailability =>
+  (dispatch, state, client) => {
+    const tenantId = employeeAvailability.tenantId;
+    const shiftId = employeeAvailability.id;
+    return client.delete<boolean>(`/tenant/${tenantId}/employee/availability/${shiftId}`).then(isSuccess => {
+      if (isSuccess) {
+        dispatch(alert.showSuccessMessage("removeAvailability", { 
+          employeeName: employeeAvailability.employee.name,
+          startDateTime: moment(employeeAvailability.startDateTime).format("LLL"),
+          endDateTime: moment(employeeAvailability.endDateTime).format("LLL") 
+        }));
+        dispatch(refreshShiftRoster());
+        dispatch(refreshAvailabilityRoster());
```

Comment:
Not possible, since we include indictments inside the shift and availability objects. If you add an availability, it can change the indictments of shifts (likewise for adding/updating a shift). If there were no indictments, then it would be possible.

Reply:
(note this is only for when a client changes/update a shift/availability)